airflow/models/dag.py

```diff
@@ -2280,6 +2284,12 @@ def create_dagrun(
                 "Creating DagRun needs either `run_id` or both `run_type` and `execution_date`"
             )
 
+        if not self.allow_manual and run_type == DagRunType.MANUAL:
+            raise AirflowException(
+                f"Cannot manually trigger DAG {self.dag_id}, because 'allow_manual' is "
+                f"set to {self.allow_manual}"
+            )
```

Comment:
Use a dedicated exception subclass here. You will need to catch this in a lot of places, and using `AirflowException` directly will make things very difficult.